nltepop.cc

```diff
@@ -531,25 +542,43 @@ void nltepop_matrix_add_ionisation(const int nonemptymgi, const int element, con
 }
 
 void nltepop_matrix_add_nt_ionisation(const int nonemptymgi, const int element, const int ion,
-                                      const std::vector<double> &s_renorm, gsl_matrix *rate_matrix_ntcoll_bf) {
+                                      const std::vector<double> &s_renorm, gsl_matrix *rate_matrix_ntcoll_bf,
+                                      const int first_ion_used, const int nions_used) {
   // collisional ionization by non-thermal electrons
 
-  assert_always(ion + 1 < get_nions(element));  // can't ionise the top ion
+  assert_always(ion + 1 < nions_used + first_ion_used);  // updated to cover case when ions have been stripped
```

Comment:
As above, the old comment made more sense to me (and brackets would be helpful). If you think the new comment is better, please explain why.